.github/workflows/codeql-analysis.yml

```
@@ -0,0 +1,46 @@
+name: "CodeQL Analysis"
+
+on:
+  push:
+    paths:
+      - "examples/java/**"
+      - ".github/workflows/codeql-analysis.yml"
+  pull_request:
+    paths:
+      - "examples/java/**"
+      - ".github/workflows/codeql-analysis.yml"
+  schedule:
+    - cron: '36 12 * * 0'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: examples/java
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'java' ]
```

To prevent diff noise in the future, I recommend avoiding inline lists
```suggestion
        language:
        - 'java'
```